kong/db/migrations/core/015_270_to_280.lua

```diff
@@ -0,0 +1,122 @@
+-- remove repeated targets, the older ones are not useful anymore. targets with
+-- weight 0 will be kept, as we cannot tell which were deleted and which were
+-- explicitly set as 0.
```

Comment:
Since the target history removal in 2.2 we don't keep deleted targets.

Reply:
Thanks! I saw kong has run `c_remove_unused_targets` in `012_213_to_220.lua` to delete duplicate targets in 2.2.

But duplicate targets may still exist after that.
Regarding https://github.com/Kong/kong/blob/master/kong/db/dao/targets.lua#L50-L59
kong will do uniqueness detection and then insert targets. 
It is not an atomic operation and a duplicate target may be added during the above process.